substrate/client/transaction-pool/src/fork_aware_txpool/view_store.rs

```diff
@@ -338,6 +338,40 @@ where
 		self.active_views.read().is_empty() && self.inactive_views.read().is_empty()
 	}
 
+	/// Searches in the view store for the first descendant view by iterating through the fork of
+	/// the `at` block, up to the provided `block_number`.
+	///
+	/// Returns with a maybe pair of a view and a set of enacted blocks when the first view is
+	/// found.
+	pub(super) fn find_view_descendent_upto_number(
+		&self,
+		at: &HashAndNumber<Block>,
+		block_number: <<Block as BlockT>::Header as Header>::Number,
+	) -> Option<(Arc<View<ChainApi>>, Vec<Block::Hash>)> {
+		let mut enacted_blocks = Vec::new();
+		let mut at_hash = at.hash;
+		let mut at_number = at.number;
+
+		// Search for a view that can be used to get and return an approximate ready
+		// transaction set.
+		while at_number > block_number {
```

Comment:
```suggestion
		while at_number >= block_number {
```

Otherwise you are not taking the finalized view?

Reply:
[a9a337d](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/a9a337d7cc2cbb2332c24eaec4b8517ecd22aea0)